src/Tasks/Microsoft.NET.Build.Tasks/ResolveCopyLocalAssets.cs

```diff
@@ -74,31 +67,30 @@ protected override void ExecuteCore()
 
             projectContext.PackagesToBeFiltered = packagestoBeFiltered;
 
-            var assemblyResolver =
-                new PublishAssembliesResolver(packageResolver)
-                    .WithExcludeFromPublish(excludeFromPublishPackageIds)
+            var fileResolver =
+                new FileResolver(NuGetPackageResolver.CreateResolver(lockFile, ProjectPath))
+                    .WithExcludedPackages(PackageReferenceConverter.GetPackageIds(ExcludedPackageReferences))
                     .WithPreserveStoreLayout(PreserveStoreLayout);
 
-            IEnumerable<ResolvedFile> resolvedAssemblies = assemblyResolver.Resolve(projectContext);
-            foreach (ResolvedFile resolvedAssembly in resolvedAssemblies)
+            foreach (var resolvedFile in fileResolver.Resolve(projectContext))
             {
-                TaskItem item = new TaskItem(resolvedAssembly.SourcePath);
-                item.SetMetadata("DestinationSubPath", resolvedAssembly.DestinationSubPath);
-                item.SetMetadata("AssetType", resolvedAssembly.Asset.ToString().ToLower());
-                item.SetMetadata(MetadataKeys.PackageName, resolvedAssembly.Package.Id.ToString());
-                item.SetMetadata(MetadataKeys.PackageVersion, resolvedAssembly.Package.Version.ToString().ToLower());
-
-                if (resolvedAssembly.Asset == AssetType.Resources)
+                TaskItem item = new TaskItem(resolvedFile.SourcePath);
+                item.SetMetadata("DestinationSubPath", resolvedFile.DestinationSubPath);
```

Comment:
What's the difference between DestinationSubPath and DestinationSubDirectory?  This should be explained somewhere, if it isn't already.

Reply:
I'll add a comment where appropriate.  The former is a sub path to the file and the latter is just the sub path to the directory containing the file.